File: clients/PhoenixLogger.php
<?php


class PhoenixLogger{
    /**
     * Set MAX_MESSAGE_LENGTH to accomodate upto 1MB of Data.
     */
    const MAX_MESSAGE_LENGTH = 1048576;
    public $socket;
    function __construct($host, $port, $serviceName){
        $this->socket = socket_create(AF_INET, SOCK_STREAM, 0) or die("Failed to create Socket.");
        socket_connect($this->socket, $host, $port) or die("Failed to connect to $host:$port");
        socket_read($this->socket, self::MAX_MESSAGE_LENGTH);
        socket_write($this->socket, $serviceName, strlen($serviceName));
    }

    function log($object){
        socket_write($this->socket, $object, strlen($object)) or die('Failed to send Log.');
    }
}

?>

File: clients/PhoenixLogger.java
import  java.net.*;
import  java.io.*;

public class PhoenixLogger {
    Socket socket = null;
    InputStream inputStream = null;
    OutputStream outputStream = null;

    public PhoenixLogger(String host, int port, String serviceName) {
        try{
            /** Connect to Log Server */
            socket = new Socket(host, port);

            /**
             * Initialize Input Stream and Read incoming data from server.
             * Just read the input no need to take any action
             * */
            inputStream = socket.getInputStream();
            inputStream.read(new byte[1024]);

            /**  Prepare Output Stream and send serviceName to log server. */
            outputStream = socket.getOutputStream();
            outputStream.write(serviceName.getBytes("UTF8"));
        }catch (Exception e){
            System.out.println(e.getMessage());
        }
    }

    void log(String object){
        if(socket.isConnected()){
            try{
                outputStream.write(object.getBytes("UTF8"));
            }catch (Exception e){
                System.out.println(e.getMessage());
            }
        }
    }

    void closeConnection(){
        try{
            socket.close();
            inputStream.close();
            outputStream.close();
        }catch (Exception e){
            System.out.println(e.getMessage());
        }
    }

}
